smithy-model/src/main/java/software/amazon/smithy/model/validation/validators/HttpBindingsMissingValidator.java

```diff
@@ -41,29 +43,44 @@
 
         TopDownIndex topDownIndex = TopDownIndex.of(model);
         return model.shapes(ServiceShape.class)
-                .flatMap(shape -> validateService(topDownIndex, shape).stream())
+                .flatMap(shape -> validateService(topDownIndex, model, shape).stream())
                 .collect(Collectors.toList());
     }
 
-    private List<ValidationEvent> validateService(TopDownIndex topDownIndex, ServiceShape service) {
+    private List<ValidationEvent> validateService(TopDownIndex topDownIndex, Model model, ServiceShape service) {
         Set<OperationShape> operations = topDownIndex.getContainedOperations(service);
         // Stop early if there are no bindings at all in the model for any operation.
         if (operations.stream().noneMatch(this::hasBindings)) {
             return ListUtils.of();
         }
 
+        Severity severity = determineSeverity(service, model);
+
         return operations.stream()
                 .filter(shape -> !shape.getTrait(HttpTrait.class).isPresent())
-                .map(shape -> createEvent(service, shape))
+                .map(shape -> createEvent(severity, service, shape))
                 .collect(Collectors.toList());
     }
 
+    // Only emit ERROR events if any of the protocols for the service support HTTP traits
+    private Severity determineSeverity(ServiceShape service, Model model) {
+        if (service.getAllTraits().values()
```

Comment:
You could use https://github.com/awslabs/smithy/blob/main/smithy-model/src/main/java/software/amazon/smithy/model/knowledge/ServiceIndex.java#L78 instead